authentication.js

```diff
@@ -23,7 +23,12 @@ module.controller('LoginController', ['$scope', '$rootScope', '$window', '$locat
   });
 
   $scope.$on('dispatchbot.authentication.failure', function (event, data) {
-    $scope.message = 'Error: Invalid user or password';
+    if (data.data.organization_key) {
+      $scope.$parent.patronRedirect = data.data.organization_key
```

Comment:
To make this variable more generic, lets call it something like 'desiredOrganizationKey' or something to that affect.